app/service/visitor-service.js

```diff
@@ -53,7 +53,7 @@ angular.module('museumApp')
     let nextRoom = visitor.location[direction];
     if(nextRoom){
       visitor.location = mapLayout[nextRoom];
-      visitor.history(visitor.location);
+      visitor.history.push(visitor.location);
```

Comment:
You should be using the ``visitory.pushHistory`` function that you wrote a few lines above because that actually makes the most recent move appear at the top using JavaScript's built-in ``unshift`` method which prepends an element to beginning of an array. 

Line 56 is adding your most recent move to the bottom of the list. Not losing points since this wasn't explicitly said, but just FYI. :)